org.osgi.test.cases.typedevent/src/org/osgi/test/cases/typedevent/junit/SingleLevelWildcardTest.java

```diff
@@ -0,0 +1,426 @@
+					dictionaryOf(TypedEventConstants.TYPED_EVENT_TOPICS,
+							"+/bar/baz"));
+
+			Map<String,Object> event = new HashMap<>();
+			event.put("test", "value");
+
+			eventBus.deliverUntyped("foo/bar/baz", event);
+
+			Mockito.verify(handler, Mockito.timeout(1000))
+					.notifyUntyped(eq("foo/bar/baz"), eq(event));
+		}
+
+		@Test
+		@DisplayName("foo/bar/+ should match foo/bar/baz")
+		public void testWildcardAtEnd() {
+			UntypedEventHandler handler = mock(UntypedEventHandler.class);
+
+			context.registerService(UntypedEventHandler.class, handler,
+					dictionaryOf(TypedEventConstants.TYPED_EVENT_TOPICS,
+							"foo/bar/+"));
+
+			Map<String,Object> event = new HashMap<>();
+			event.put("test", "value");
+
+			eventBus.deliverUntyped("foo/bar/baz", event);
+
+			Mockito.verify(handler, Mockito.timeout(1000))
+					.notifyUntyped(eq("foo/bar/baz"), eq(event));
+		}
+
+		@Test
+		@DisplayName("foo/+/baz should match foo/bar/baz")
+		public void testWildcardInMiddle() {
+			UntypedEventHandler handler = mock(UntypedEventHandler.class);
+
+			context.registerService(UntypedEventHandler.class, handler,
+					dictionaryOf(TypedEventConstants.TYPED_EVENT_TOPICS,
+							"foo/+/baz"));
+
+			Map<String,Object> event = new HashMap<>();
+			event.put("test", "value");
+
+			eventBus.deliverUntyped("foo/bar/baz", event);
+
+			Mockito.verify(handler, Mockito.timeout(1000))
+					.notifyUntyped(eq("foo/bar/baz"), eq(event));
+		}
+	}
+
+	@DisplayName("Special Case: + Alone")
+	@Nested
+	class PlusAlone {
+
+		@Test
+		@DisplayName("+ should match all single-level topics")
+		public void testPlusMatchesSingleLevel() {
+			UntypedEventHandler handler = mock(UntypedEventHandler.class);
+
+			context.registerService(UntypedEventHandler.class, handler,
+					dictionaryOf(TypedEventConstants.TYPED_EVENT_TOPICS, "+"));
+
+			Map<String,Object> event = new HashMap<>();
+			event.put("test", "value");
+
+			eventBus.deliverUntyped("foo", event);
+
+			Mockito.verify(handler, Mockito.timeout(1000))
+					.notifyUntyped(eq("foo"), eq(event));
+		}
+
+		@Test
+		@DisplayName("+ should not match multi-level topics")
+		public void testPlusDoesNotMatchMultiLevel() {
+			UntypedEventHandler handler = mock(UntypedEventHandler.class);
+
+			context.registerService(UntypedEventHandler.class, handler,
+					dictionaryOf(TypedEventConstants.TYPED_EVENT_TOPICS, "+"));
+
+			Map<String,Object> event = new HashMap<>();
+			event.put("test", "value");
+
+			eventBus.deliverUntyped("foo/bar", event);
+
+			Mockito.verify(handler, Mockito.after(1000).never())
+					.notifyUntyped(any(), any());
+		}
+	}
+
+	@DisplayName("Typed Event Handlers with Single-Level Wildcards")
+	@Nested
+	class TypedHandlerWildcards {
+
+		@SuppressWarnings("unchecked")
+		@Test
+		@DisplayName("Typed handler should receive events matching single-level wildcard")
+		public void testTypedHandlerWithSingleLevelWildcard() {
+			TypedEventHandler<EventA> handler = mock(TypedEventHandler.class);
```

Comment:
```suggestion
			TypedEventHandler<EventA> handler = mock(TypedEventHandlerA.class);
```

Reply:
This test was failing as mockito doesn't add generic type information to its implemented interfaces. This means that the event handler service is invalid without either a bound type (e.g. by using `TypedEventHandlerA`) or a service property saying that you want to use `EventA` as the type for the handler.